### DISCUSSION TITLE
Welcome to firmata4j Discussions!

### DISCUSSION THREAD

**Discussion by kurbatov** [kurbatov/firmata4j#46] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you build with **firmata4j**.

**Comment by onnimikki** [comment]
Encountered an error at runtime under Windows 10.  Didn't have this issue with macOS.

```
# JRE version: OpenJDK Runtime Environment Temurin-16.0.2+7 (16.0.2+7) (build 16.0.2+7)
# Java VM: OpenJDK 64-Bit Server VM Temurin-16.0.2+7 (16.0.2+7, mixed mode, sharing, tiered, compressed oops, compressed class ptrs, g1 gc, windows-amd64)
# Problematic frame:
# C  [jSSC-2.8_x86_64.dll+0xb5db]
```

It seems that jSSC is having an issue.  Do you have a recommendation for resolving this issue?

**Reply by kurbatov** [reply]
Hi James,

Windows 10 has issues with jSSC.

There is an implementation of the serial transport that relies on the `jSerialComm` library. This implementation is available in `feat/serial-transport` branch of this repository. In order to try it out, please, clone this repo, checkout `feat/serial-transport` branch and build the library locally:

```bash
$ git clone git@github.com:kurbatov/firmata4j.git
$ cd ./firmata4j
$ git checkout feat/serial-transport
$ mvn clean install
```

After that you can use the locally built library in a project of yours declaring the following dependencies:

```xml
<dependency>
   <groupId>com.github.kurbatov</groupId>
    <artifactId>firmata4j</artifactId>
    <version>2.3.9-SNAPSHOT</version>
</dependency>
<dependency>
    <groupId>com.fazecast</groupId>
    <artifactId>jSerialComm</artifactId>
    <version>2.6.2</version>
</dependency>
```

The serial transport implementation in this branch selects underlying library from the libraries in the classpath of the application.

Please, let me know if it works for you. If yes, I merge this branch in the mainline and it will be ready for release.

**Reply by kurbatov** [reply]
As an option, it is worth trying a newer version of jSSC.

```xml
        <dependency>
            <groupId>io.github.java-native</groupId>
            <artifactId>jssc</artifactId>
            <version>2.9.4</version>
        </dependency>
```

I have not tried it myself, just noticed that someone uses a fork of this repo with this version.
